library-best-practices/JLBP-16.md

```diff
@@ -0,0 +1,28 @@
+[JLBP-16] Produce a BOM for multi-module projects
+-------------------------------------------------
+
+### The BOM (Bill of Materials)
+
+- A BOM is a `pom.xml` file that has a `<dependencyManagement>` section that lists all the modules of a project. It does not have `<modules>` or `<dependencies>` elements itself. It also has `<type>pom</type>` declared.
+  - All of the modules of the project should be included, otherwise a consumer trying to use an unspecified module will have to manually specify and upgrade the version of that module, defeating the purpose of the BOM.
+- The purpose of a BOM is to allow consumers of a library to ensure that they are using a consistent set of versions for artifacts released by a single project. When imported, a BOM operates as a filter. It does not cause any modules to be added as dependencies. Instead, for any dependency in a project's dependency tree, *if* that module appears in the BOM, the version from the BOM is used.
```

Comment:
"enable consumers of a library to select a consistent set of versions"

Reply:
Will do.